test/e2e-go/e2e_go_suite_test.go

```diff
@@ -59,43 +52,17 @@ var _ = BeforeSuite(func(done Done) {
 	By("creating a new directory")
 	Expect(tc.Prepare()).To(Succeed())
 
-	By("checking the cluster type")
-	kubectx, err = tc.Kubectl.Command("config", "current-context")
+	By("fetching the current-context")
+	tc.Kubectx, err = tc.Kubectl.Command("config", "current-context")
```

Comment:
Can this be centralized and re-located into `NewTestContext()`?

Reply:
it's already centralized by being in the suite, centralizing further would be more DRY but require more lines - should the same be said for all the lines that are shared between the different e2e suites?

Reply:
It's fewer lines no matter how you cut it right?

Several calls like this in different suites (e.g. go, ansible, helm)
```go
By("fetching the current-context")
tc.Kubectx, err = tc.Kubectl.Command("config", "current-context")
Expect(err).NotTo(BeNil())
```

Would be replaced with one call in `NewTestContext()`:

```go
if tc.Kubectx, err = tc.Kubectl.Command("config", "current-context"); err != nil {
    return nil, err
}
```

At this point, we could also unexport `tc.Kubectx` since callers interact with the value of `tc.Kubectx` through the `tc.IsRunningOnKind` function.

That also makes me wonder, can `IsPrometheusManagedBySuite` and `IsOLMManagedBySuite` also be unexported?

Reply:
Hi @joelanford, 
 
> Can this be centralized and re-located into NewTestContext()?

I tried that before and did not work well.

```
  Unexpected error:
      <*errors.errorString | 0xc0001c4d30>: {
          s: "kubectl config current-context failed with error: ",
      }
      kubectl config current-context failed with error: 
  occurred
```

Reply:
Hi @joelanford, 

I think we can get this one merged and then after we can check if has a better way to centralize the `kubectl config current-context` call. However, but I think it is not a reason to block this one. WDYT? do you agree with that?
Also, see that the other suggestions made by you are addressed.

Reply:
Yep! My comment was a nit and can be investigated more in a follow up.